Multi version ext/pg partman

samrose opened the pull request:
refactor of pg_partman to include multiversion support, switching script, update of tests

PSQL-303

samrose committed:
feat: pg_partman all major versions

encima committed:
add tests for pg_partman

encima committed:
add tests for pg_partman

encima committed:
add partman to local tests. remove from initial schema

encima committed:
add hypertable to partman test

samrose committed:
wip: stage pg_partman multi-version changes

samrose committed:
feat: multiversion pg_partman

encima committed:
update v15 schema

jfroche committed:
Add nixos test for pg_partman extension

Run upgrade and pg_regress tests for pg_partman extension in NixOS tests.

encima commented:
[pg_partman](https://github.com/pgpartman/pg_partman/releases/tag/v5.3.1) has a 5.3.1 release that could be worth moving to before merging (solves some uuidv7 things and better 17/18 compat)

encima committed:
add 5.3.1 version for partman

jfroche committed:
fix: remove old pg_partman version and fix pg_regress outputs

encima commented:
@jfroche was/is there a reason to rename the background worker for the shared library? It is likely best to keep the path the same so it does not deviate from the docs

jfroche committed:
fix(pg_partman): use correct library name for background worker

The pg_partman extension uses a different library name (pg_partman_bgw) than its extension name (pg_partman). This change separates the library name from the extension name to properly handle the background worker library installation and version switching.

jfroche commented:
> @jfroche was/is there a reason to rename the background worker for the shared library? It is likely best to keep the path the same so it does not deviate from the docs

I didn't realize the project was using different names for the library and the extension. I have adapted the code for that in https://github.com/supabase/postgres/pull/1854/commits/d199683aa69c1f80eb99536763c531487258e624

encima commented:
Tested:
- [x] Studio shows extension
- [x] Upgrade from 15 to 17
- [x] Read Replicas

<img width="1518" height="341" alt="image" src="https://github.com/user-attachments/assets/ebfda680-3924-4c85-9dc9-e7eb79941f54" />

samrose committed:
chore: bump again to release
